### PR TITLE
fix: fix Invalid URL error when using uppercase i18n url

### PR DESCRIPTION
**Pull request by sarahetter**
### Summary

When adding a locale such as `de-DE` to next.config.ts, we were giving an Invalid URL error as it our cdn lowercases paths. By adding a rewrite when these locale redirects are happening, we are able to point at the correct site.

### Test plan

1. Visit the middleware  Deploy Preview `/de-DE/static` route
2. Observe that the page shows up without an `Invalid URL` error

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal
Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/300
Screen Shot 2022-11-24 at 2 59 27 PM

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

File: jestSetup.js [hunk 1]
```diff
@@ -1,2 +1,2 @@
-// eslint-disable-next-line n/no-unpublished-require
+
```
Comment (sarahetter) [inline jestSetup.js:1]: This is happening when I run `npm i` in the package 😕

File: packages/next/src/middleware/request.ts [hunk 2]
```diff
@@ -60,7 +56,16 @@ export class MiddlewareRequest extends Request {
    */
   async next(options?: NextOptions): Promise<MiddlewareResponse> {
     this.applyHeaders()
-    const response = await this.context.next(options)
+    let response = await this.context.next(options)
```
Comment (nickytonline) [inline packages/next/src/middleware/request.ts:59]: [dust] I imagine this can stay as `const`.

**Review by nickytonline — APPROVED** [review]
Looks good. Left a dusty comment.

File: jestSetup.js (hunk 1 above)
Comment (nickytonline) [reply jestSetup.js:1]: Does it cause a lint error? If not, it's fine that it's removed.

Comment (sarahetter) [reply jestSetup.js:1]: No more lint errors than are currently in main :D

File: packages/next/src/middleware/request.ts [hunk 3]
```diff
@@ -60,7 +56,16 @@ export class MiddlewareRequest extends Request {
...
+
+    // next i18n responses contain a relative location header, and a 301 status
```
Comment (ascorbic) [inline packages/next/src/middleware/request.ts:61]: [dust] this isn't specifically a Next i18n thing: this is our CDN redirecting to a lowercase version of any static file request. In this scenario the i18n file is statically-generated, so the cdn redirects to the lowercase version

File: packages/next/src/middleware/request.ts (hunk 2 above)
Comment (sarahetter) [reply packages/next/src/middleware/request.ts:59]: It can't as it's potentially getting redefined below in the if statement.

Comment (nickytonline) [reply packages/next/src/middleware/request.ts:59]: Ahh, didn't see that. 🙃

**Comment by sarahetter** [comment]
@ascorbic / @nickytonline Just looked with Nick - PR looks fine, just a few of the deploy previews are failing because it's unable to find the new `@netlify/edge-functions` (and some are able to find it properly 😕 ) 

I'm away for December, can one of you take care of this while I'm gone?

**Comment by ascorbic** [comment]
I think I know what's up. I'll take a look

**Review by ascorbic — APPROVED** [review]

**Pull request merged**